### DISCUSSION TITLE
Template File to Database Sync Ability

### DISCUSSION THREAD

**Discussion by ccbangler** [ExpressionEngine/ExpressionEngine#2957] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
When working direct with template files outside of EE including renaming or removing them, the database is not synced with the changes and the files still exist in the Db form and can be accessed through live EE pages. 

**Describe the solution you'd like**
The ability to sync templates from currently existing files to the Database and vice-versa though a Utility in the EE Control Panel This would remove non-existing templates and update currently existing template data as needed depending on the option selected. This functionality was included previously (EE 2x) in part and since has been removed due to being a one way sync and had caused some developer confusion. 

**Describe alternatives you've considered**
The only way to manage this currently is to delete and update all the files manually through the EE Template Manager which is unreasonable with regards to large website redevelopment projects.

**Teachability, Documentation, Adoption, Migration Strategy**
Caution Box: Syncing templates will overwrite existing template data and/or files and may produce undesirable results. Please use this feature with the upmost caution. 

Step 1. Backup Expression Engine Data and Template Files

Step 2. Select the Template Sync direction you are attempting to perform. 
Directions Radio Button
Option 1. File(s) to Database
This option will replace or remove all templates in the database with the physical template files in System > User > Templates 
This cannot be undone.

Option 2. Database to File(s)
This option will replace or remove all template files in System > User > Templates that do not match or are not in the database  respectively.
This cannot be undone.

Step 3. Sync Option(s)
Checkbox
1. Select to only Sync New and Updated Templates
This option will only update the Files and Database based on direction above that are newer using the database timestamp vs. file edited timestamp.
 
2. Warn before deleting Template files or Template data
With this option the EE will evaluate the sync before proceeding and show a warning the user must accept before continuing with changes

3. Download New Template Backup after Template Sync Completes

Submit > Are you sure? modal (one more attempt to protect people from themselves)

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
Just noting that it's likely related to https://github.com/ExpressionEngine/ExpressionEngine/issues/901 and https://github.com/ExpressionEngine/ExpressionEngine/issues/532
